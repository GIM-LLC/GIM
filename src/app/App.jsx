import React from 'react';
import './app.css';
import { SocketContext, socket } from '../context/SocketProvider';
import CursorWrapper from '../cursors/CursorWrapper';

export default function App() {
  return (
    <SocketContext.Provider value={socket}>
      <CursorWrapper />
    </SocketContext.Provider>
  );
}
