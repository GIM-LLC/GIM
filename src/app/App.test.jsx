/* eslint-disable indent */
import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('GIM app testing', () => {
    it('checks that main components did mount on initial load', async () => {
        const gim = render(<App />);

        //check that nav bar in header is there 
        const navList = screen.getByRole('list', { name: 'navList' });
        expect(navList).not.toBeEmptyDOMElement();

        //check that home page section includes three images
        const galleryImages = screen.getAllByAltText('cage');
        expect(galleryImages).toHaveLength(3);

        //check that footer is there
        const footerList = screen.getByRole('list', { name: 'footerList' });
        expect(footerList).not.toBeEmptyDOMElement();

    });
});
