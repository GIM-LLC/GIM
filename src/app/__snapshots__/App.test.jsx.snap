// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIM app testing checks that main components did mount on initial load 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      
      
      <div
        class="cursorWrapper"
        id="cursorWrapper"
      >
        <header
          class="header"
        >
          <span
            class="masthead"
          >
            <h1
              class="gim"
              style="color: rgb(58, 58, 58);"
            >
              G.I.M.
            </h1>
            <h3>
              -LLC.-
            </h3>
          </span>
          <img
            alt="dani-duck"
            class="hidden"
            src="/assets/dani-duck.png"
          />
          <nav
            class="navBar"
          >
            <ul
              aria-label="navList"
            >
              <li
                class="navLink"
              >
                <a
                  href="#"
                  style="color: rgb(63, 138, 114);"
                >
                  about
                </a>
              </li>
              <li
                class="navLink"
              >
                <a
                  href="#"
                  style="color: rgb(63, 138, 114);"
                >
                  locations
                </a>
              </li>
              <li
                class="navLink"
              >
                <a
                  href="#"
                  style="color: rgb(63, 138, 114);"
                >
                  join us
                </a>
              </li>
              <li
                class="navLink"
              >
                <a
                  href="#"
                  style="color: rgb(63, 138, 114);"
                >
                  press
                </a>
              </li>
              <li
                class="navLink"
              >
                <a
                  href="#"
                  style="color: rgb(63, 138, 114);"
                >
                  investors
                </a>
              </li>
            </ul>
            <section>
              <form>
                <input
                  class=""
                  placeholder="Search Term"
                  type="text"
                  value=""
                />
                <button>
                  SEARCH
                </button>
              </form>
            </section>
          </nav>
        </header>
        <main
          class="homePage shakeElement"
        >
          <section
            class="companyProfile"
          >
            <article
              class="story"
            >
              <figure>
                <img
                  alt="gim image"
                  class="objectPulse"
                  src="/assets/thumbs1.jpg"
                />
              </figure>
              <article>
                <h2>
                  The GIM Story
                </h2>
                <p>
                  General Industrial Manufacturing Company is a three-generation, family-owned and operated, precision machining supplier. GIM has earned a reputation as a supplier of high quality, low cost precision parts with on-time deliveries. We are a top quality team of employees who work well together.
                </p>
              </article>
            </article>
            <section
              class="news"
            >
              <h3>
                Company News
              </h3>
              <article>
                <h6>
                  Manufacturing Leadership Award Winner 2021 - 06/04/21
                </h6>
                <p>
                  This marks the 3rd time GIM has accepted this award since our start in 1962.
                </p>
              </article>
              <article>
                <h6>
                  Company Expansion Plan Revealed - 05/22/21
                </h6>
                <p>
                  Plans to expand company reach and hire 100 new employees to start in 2022.
                </p>
              </article>
              <article
                class="objectPulse"
              >
                <h6>
                  Employee Memorial Fundraiser - 04/01/21
                </h6>
                <p>
                  With the unexpected passing of one of our long time employees,
                   
                  <span
                    class="name"
                  >
                    Robin Smith
                  </span>
                  , HR has organized a fundraiser to help the family of our late team member.
                </p>
              </article>
            </section>
          </section>
          <section
            class="gallery"
          >
            <div
              class="container"
            >
              <figure
                class="captionedImage"
              >
                <img
                  alt="cage"
                  src="/assets/networking1.jpg"
                />
              </figure>
              <div
                class="imageOverlay"
              >
                <div
                  class="text"
                >
                  <h3>
                    Stockholder Information
                  </h3>
                  <div>
                    <p>
                      Stockholders Dividend are paid quarterly on the 5th business day of June, August, October, and November at end of day. Stockholders may have their funds deposited via check and direct deposit.
                    </p>
                    <button
                      class=""
                    >
                      View More
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="container"
            >
              <figure
                class="captionedImage"
              >
                <img
                  alt="cage"
                  src="/assets/helpdesk1.jpg"
                />
              </figure>
              <div
                class="imageOverlay"
              >
                <div
                  class="text"
                >
                  <h3>
                    Employment
                  </h3>
                  <div>
                    <p>
                      Supporting you as you make the next step in your career towards state of the art manufacturing technology. Dozens of open roles at your fingertips to help shape the future of industrialized manufacturing. For contracting inquiries please contact via email.
                    </p>
                    <button
                      class=""
                    >
                      View More
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="container objectPulse"
            >
              <figure
                class="captionedImage"
              >
                <img
                  alt="cage"
                  src="/assets/collaborate1.jpg"
                />
              </figure>
              <div
                class="imageOverlay"
              >
                <div
                  class="text"
                >
                  <h3>
                    Life with GIM
                  </h3>
                  <div>
                    <p>
                      Here at GIM you will find minimal bureaucracy, and more support. GIM has a welcoming and warm culture, ready to serve the community with amazing industrial manufacturing products
                    </p>
                    <button
                      class=""
                    >
                      View More
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </section>
          <section
            class="mission"
          >
            <p
              class="missionHeader"
            >
              Core Values
            </p>
            <div
              class="missionBody"
            >
              <p>
                Her
                <span
                  class=""
                >
                  e
                </span>
                 at General Industrial Manufacturers, we believe in providing the best quality service and products to our loyal customer base. We strive to create economic growth through infrastructure and energy development, and to provide 
                <span
                  class=""
                >
                  s
                </span>
                olutions that support communities and protect the planet. Our goal is to provide the highest level of service, the broadest selection of produ
                <span
                  class=""
                >
                  c
                </span>
                ts and the most competitive prices.
              </p>
              <p>
                The strength of our team does not only lie in our combined experience and expertise but our 
                <span
                  class=""
                >
                  a
                </span>
                bility to trust each other, 
                <em>
                  no matter what the circumstances
                </em>
                . We know that our best work is not produced by individuals but by enjoying collaboration as a team and su
                <span
                  class=""
                >
                  p
                </span>
                porting each other. This team mentality also extends throughout every day as we work collaboratively with each other to m
                <span
                  class=""
                >
                  e
                </span>
                et our goals.
              </p>
            </div>
          </section>
          <p>
            <img
              alt="ghostie"
              class="ghost positionOne"
              src="/assets/yellow-ghostie.png"
            />
          </p>
        </main>
        <div>
          <div
            class="chatIcon"
          >
            <span>
              <i
                class="far fa-comment-dots"
              />
            </span>
          </div>
        </div>
        <footer
          class="footer"
        >
          <nav>
            <ul
              aria-label="footerList"
            >
              <li
                class="footerListItem"
              >
                <p
                  class="headers"
                >
                  our people
                </p>
                <p>
                  Meet the Suppliers
                </p>
                <p>
                  The Journey
                </p>
                <p>
                  Leadership
                </p>
                <p>
                  Experts in the field
                </p>
              </li>
              <li
                class="footerListItem"
              >
                <p
                  class="headers"
                >
                  why GIM
                </p>
                <p
                  class="pointer"
                >
                  Transparency
                </p>
                <p>
                  Commitment to Sustainability
                </p>
                <p>
                  Contributions & Partnerships
                </p>
                <p>
                  Global Health Impact
                </p>
              </li>
              <li
                class="footerListItem"
              >
                <p
                  class="headers"
                >
                  our products
                </p>
                <p>
                  Raw Materials
                </p>
                <p>
                  Installations
                </p>
                <p>
                  Accesories
                </p>
                <p>
                  Supplies and Consumables
                </p>
              </li>
              <li
                class="footerListItem"
              >
                <p
                  class="headers"
                >
                  contact us
                </p>
                <div
                  class="contactUsDiv"
                >
                  <div
                    class="iconDiv"
                  >
                    <img
                      alt="twitter-icon"
                      class="iconImage"
                      src="/assets/Twitter.png"
                    />
                    <img
                      alt="instagram-icon"
                      class="iconImage"
                      src="/assets/ig.png"
                    />
                    <img
                      alt="github-icon"
                      class="iconImage"
                      src="/assets/Github_icon.png"
                    />
                    <img
                      alt="linkedin-icon"
                      class="iconImage"
                      src="/assets/linkedin-icon-2.png"
                    />
                    <img
                      alt="ghost-icon"
                      class="iconImageGhost pulse"
                      src="/assets/ghostie.png"
                    />
                  </div>
                  <div
                    class="addressDiv"
                  >
                    <p
                      class="addressesHeader"
                    >
                      Corporate Offices
                    </p>
                    <p
                      class="addresses"
                    >
                      Oak Square Business Park
                    </p>
                    <p
                      class="addresses"
                    >
                      1532 Smithsonian St
                    </p>
                    <p
                      class="addresses"
                    >
                      Middleton, Nebraska, 68833
                    </p>
                  </div>
                </div>
              </li>
            </ul>
          </nav>
          <div
            class="hiddenMessage hide"
          >
            YES! The answers are all here.... keep clicking around!
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    
    
    <div
      class="cursorWrapper"
      id="cursorWrapper"
    >
      <header
        class="header"
      >
        <span
          class="masthead"
        >
          <h1
            class="gim"
            style="color: rgb(58, 58, 58);"
          >
            G.I.M.
          </h1>
          <h3>
            -LLC.-
          </h3>
        </span>
        <img
          alt="dani-duck"
          class="hidden"
          src="/assets/dani-duck.png"
        />
        <nav
          class="navBar"
        >
          <ul
            aria-label="navList"
          >
            <li
              class="navLink"
            >
              <a
                href="#"
                style="color: rgb(63, 138, 114);"
              >
                about
              </a>
            </li>
            <li
              class="navLink"
            >
              <a
                href="#"
                style="color: rgb(63, 138, 114);"
              >
                locations
              </a>
            </li>
            <li
              class="navLink"
            >
              <a
                href="#"
                style="color: rgb(63, 138, 114);"
              >
                join us
              </a>
            </li>
            <li
              class="navLink"
            >
              <a
                href="#"
                style="color: rgb(63, 138, 114);"
              >
                press
              </a>
            </li>
            <li
              class="navLink"
            >
              <a
                href="#"
                style="color: rgb(63, 138, 114);"
              >
                investors
              </a>
            </li>
          </ul>
          <section>
            <form>
              <input
                class=""
                placeholder="Search Term"
                type="text"
                value=""
              />
              <button>
                SEARCH
              </button>
            </form>
          </section>
        </nav>
      </header>
      <main
        class="homePage shakeElement"
      >
        <section
          class="companyProfile"
        >
          <article
            class="story"
          >
            <figure>
              <img
                alt="gim image"
                class="objectPulse"
                src="/assets/thumbs1.jpg"
              />
            </figure>
            <article>
              <h2>
                The GIM Story
              </h2>
              <p>
                General Industrial Manufacturing Company is a three-generation, family-owned and operated, precision machining supplier. GIM has earned a reputation as a supplier of high quality, low cost precision parts with on-time deliveries. We are a top quality team of employees who work well together.
              </p>
            </article>
          </article>
          <section
            class="news"
          >
            <h3>
              Company News
            </h3>
            <article>
              <h6>
                Manufacturing Leadership Award Winner 2021 - 06/04/21
              </h6>
              <p>
                This marks the 3rd time GIM has accepted this award since our start in 1962.
              </p>
            </article>
            <article>
              <h6>
                Company Expansion Plan Revealed - 05/22/21
              </h6>
              <p>
                Plans to expand company reach and hire 100 new employees to start in 2022.
              </p>
            </article>
            <article
              class="objectPulse"
            >
              <h6>
                Employee Memorial Fundraiser - 04/01/21
              </h6>
              <p>
                With the unexpected passing of one of our long time employees,
                 
                <span
                  class="name"
                >
                  Robin Smith
                </span>
                , HR has organized a fundraiser to help the family of our late team member.
              </p>
            </article>
          </section>
        </section>
        <section
          class="gallery"
        >
          <div
            class="container"
          >
            <figure
              class="captionedImage"
            >
              <img
                alt="cage"
                src="/assets/networking1.jpg"
              />
            </figure>
            <div
              class="imageOverlay"
            >
              <div
                class="text"
              >
                <h3>
                  Stockholder Information
                </h3>
                <div>
                  <p>
                    Stockholders Dividend are paid quarterly on the 5th business day of June, August, October, and November at end of day. Stockholders may have their funds deposited via check and direct deposit.
                  </p>
                  <button
                    class=""
                  >
                    View More
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <figure
              class="captionedImage"
            >
              <img
                alt="cage"
                src="/assets/helpdesk1.jpg"
              />
            </figure>
            <div
              class="imageOverlay"
            >
              <div
                class="text"
              >
                <h3>
                  Employment
                </h3>
                <div>
                  <p>
                    Supporting you as you make the next step in your career towards state of the art manufacturing technology. Dozens of open roles at your fingertips to help shape the future of industrialized manufacturing. For contracting inquiries please contact via email.
                  </p>
                  <button
                    class=""
                  >
                    View More
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="container objectPulse"
          >
            <figure
              class="captionedImage"
            >
              <img
                alt="cage"
                src="/assets/collaborate1.jpg"
              />
            </figure>
            <div
              class="imageOverlay"
            >
              <div
                class="text"
              >
                <h3>
                  Life with GIM
                </h3>
                <div>
                  <p>
                    Here at GIM you will find minimal bureaucracy, and more support. GIM has a welcoming and warm culture, ready to serve the community with amazing industrial manufacturing products
                  </p>
                  <button
                    class=""
                  >
                    View More
                  </button>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="mission"
        >
          <p
            class="missionHeader"
          >
            Core Values
          </p>
          <div
            class="missionBody"
          >
            <p>
              Her
              <span
                class=""
              >
                e
              </span>
               at General Industrial Manufacturers, we believe in providing the best quality service and products to our loyal customer base. We strive to create economic growth through infrastructure and energy development, and to provide 
              <span
                class=""
              >
                s
              </span>
              olutions that support communities and protect the planet. Our goal is to provide the highest level of service, the broadest selection of produ
              <span
                class=""
              >
                c
              </span>
              ts and the most competitive prices.
            </p>
            <p>
              The strength of our team does not only lie in our combined experience and expertise but our 
              <span
                class=""
              >
                a
              </span>
              bility to trust each other, 
              <em>
                no matter what the circumstances
              </em>
              . We know that our best work is not produced by individuals but by enjoying collaboration as a team and su
              <span
                class=""
              >
                p
              </span>
              porting each other. This team mentality also extends throughout every day as we work collaboratively with each other to m
              <span
                class=""
              >
                e
              </span>
              et our goals.
            </p>
          </div>
        </section>
        <p>
          <img
            alt="ghostie"
            class="ghost positionOne"
            src="/assets/yellow-ghostie.png"
          />
        </p>
      </main>
      <div>
        <div
          class="chatIcon"
        >
          <span>
            <i
              class="far fa-comment-dots"
            />
          </span>
        </div>
      </div>
      <footer
        class="footer"
      >
        <nav>
          <ul
            aria-label="footerList"
          >
            <li
              class="footerListItem"
            >
              <p
                class="headers"
              >
                our people
              </p>
              <p>
                Meet the Suppliers
              </p>
              <p>
                The Journey
              </p>
              <p>
                Leadership
              </p>
              <p>
                Experts in the field
              </p>
            </li>
            <li
              class="footerListItem"
            >
              <p
                class="headers"
              >
                why GIM
              </p>
              <p
                class="pointer"
              >
                Transparency
              </p>
              <p>
                Commitment to Sustainability
              </p>
              <p>
                Contributions & Partnerships
              </p>
              <p>
                Global Health Impact
              </p>
            </li>
            <li
              class="footerListItem"
            >
              <p
                class="headers"
              >
                our products
              </p>
              <p>
                Raw Materials
              </p>
              <p>
                Installations
              </p>
              <p>
                Accesories
              </p>
              <p>
                Supplies and Consumables
              </p>
            </li>
            <li
              class="footerListItem"
            >
              <p
                class="headers"
              >
                contact us
              </p>
              <div
                class="contactUsDiv"
              >
                <div
                  class="iconDiv"
                >
                  <img
                    alt="twitter-icon"
                    class="iconImage"
                    src="/assets/Twitter.png"
                  />
                  <img
                    alt="instagram-icon"
                    class="iconImage"
                    src="/assets/ig.png"
                  />
                  <img
                    alt="github-icon"
                    class="iconImage"
                    src="/assets/Github_icon.png"
                  />
                  <img
                    alt="linkedin-icon"
                    class="iconImage"
                    src="/assets/linkedin-icon-2.png"
                  />
                  <img
                    alt="ghost-icon"
                    class="iconImageGhost pulse"
                    src="/assets/ghostie.png"
                  />
                </div>
                <div
                  class="addressDiv"
                >
                  <p
                    class="addressesHeader"
                  >
                    Corporate Offices
                  </p>
                  <p
                    class="addresses"
                  >
                    Oak Square Business Park
                  </p>
                  <p
                    class="addresses"
                  >
                    1532 Smithsonian St
                  </p>
                  <p
                    class="addresses"
                  >
                    Middleton, Nebraska, 68833
                  </p>
                </div>
              </div>
            </li>
          </ul>
        </nav>
        <div
          class="hiddenMessage hide"
        >
          YES! The answers are all here.... keep clicking around!
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
